src/api/app/models/bs_request_action_add_role.rb

```diff
@@ -1,53 +1,46 @@
 #
 class BsRequestActionAddRole < BsRequestAction
-  #### Includes and extends
-  #### Constants
-  #### Self config
-  #### Attributes
-  #### Associations macros (Belongs to, Has one, Has many)
-  #### Callbacks macros: before_save, after_save, etc.
-  #### Scopes (first the default_scope macro if is used)
-  #### Validations macros
-
-  #### Class methods using self. (public and then private)
   def self.sti_name
     :add_role
   end
 
-  #### To define class methods as private use private_class_method
-  #### private
-
-  #### Instance methods (public and then protected/private)
   def check_sanity
     super
     errors.add(:role, 'should not be empty for add_role') if role.blank?
-    return unless person_name.blank? && group_name.blank?
-    errors.add(:person_name, 'Either person or group needs to be set')
+    person_or_group_present
   end
 
-  def execute_accept(_opts)
-    object = Project.find_by_name(target_project)
-    object = object.packages.find_by_name(target_package) if target_package
-    if person_name
-      role = Role.find_by_title!(self.role)
-      object.add_user(person_name, role)
-    end
-    if group_name
-      role = Role.find_by_title!(self.role)
-      object.add_group(group_name, role)
-    end
-    object.store(comment: "add_role request #{bs_request.number}", request: bs_request)
+  def execute_accept(opts)
+    object.add_user(person_name, role_object, opts[:ignore_lock]) if person_name
+    object.add_group(group_name, role_object, opts[:ignore_lock]) if group_name
+
+    object.store(comment: "#{type} request #{bs_request.number}", request: bs_request)
   end
 
   def render_xml_attributes(node)
     render_xml_target(node)
     node.person(name: person_name, role: role) if person_name
-    return unless group_name
+    node.group(name: group_name, role: role) if group_name
+  end
 
-    node.group(name: group_name, role: role)
+  protected
+
+  def person_or_group_present
+    return unless person_name.blank? && group_name.blank?
+    errors.add(:person_name, 'Either person or group needs to be set')
   end
 
-  #### Alias of methods
+  def object
+    unless @object
+      @object = Project.find_by_name!(target_project)
+      @object.packages.find_by_name(target_package) if target_package
+    end
+    @object
+  end
+
+  def role_object
+    @role_object ||= Role.find_by_title!(role)
```

Comment:
Why do we need a method for this? This is just used two times in the same place. And it's not an expensive call.

Reply:
to overload it in set_bugowner